### PR TITLE
[EASY] Pass dtype to tensor contructor in test_neg

### LINKED ISSUE #9554: test_neg is still failing out of range UBSAN
```
19:27:43 test_neg (__main__.TestTorch) ... /var/lib/jenkins/workspace/aten/src/TH/generic/THTensorCopy.cpp:234:1: runtime error: 4.02053e+202 is outside the range of representable values of type 'int'
19:27:43     #0 0x7f24d0a9f4a4 in THIntTensor_copyDouble /var/lib/jenkins/workspace/aten/src/TH/generic/THTensorCopy.cpp:234:1
19:27:43     #1 0x7f24cfc84364 in at::CPUIntType::s_copy_(at::Tensor&, at::Tensor const&, bool) const /var/lib/jenkins/workspace/build/aten/src/ATen/CPUCopy.cpp:193:9
19:27:43     #2 0x7f24de374565 in torch::autograd::VariableType::s_copy_(at::Tensor&, at::Tensor const&, bool) const /var/lib/jenkins/workspace/torch/csrc/autograd/generated/VariableType.cpp:452:13
19:27:43     #3 0x7f24d04c3b19 in at::Type::copy_(at::Tensor&, at::Tensor const&, bool) const /var/lib/jenkins/workspace/build/aten/src/ATen/Type.cpp:104:10
19:27:43     #4 0x7f24cf552e20 in at::Tensor::copy_(at::Tensor const&, bool) /var/lib/jenkins/workspace/build/aten/src/ATen/TensorMethods.h:23:17
19:27:43     #5 0x7f24d04c50f0 in at::Type::copy(at::Tensor const&, bool) const /var/lib/jenkins/workspace/build/aten/src/ATen/Type.cpp:120:7
19:27:43     #6 0x7f24cf65e7c3 in at::native::_cast_Int(at::Tensor const&, bool) /var/lib/jenkins/workspace/aten/src/ATen/native/TensorFactories.cpp:109:1
19:27:43     #7 0x7f24d05eedb5 in at::Type::_cast_Int(at::Tensor const&, bool) const /var/lib/jenkins/workspace/build/aten/src/ATen/Type.cpp:3074:13
19:27:43     #8 0x7f24de87d177 in torch::autograd::VariableType::_cast_Int(at::Tensor const&, bool) const /var/lib/jenkins/workspace/torch/csrc/autograd/generated/VariableType.cpp:20699:23
19:27:43     #9 0x7f24dc63009c in at::Tensor::_cast_Int(bool) const /var/lib/jenkins/workspace/torch/lib/tmp_install/include/ATen/TensorMethods.h:736:19
19:27:43     #10 0x7f24dc62edca in torch::utils::dispatch_type_conversion(at::Tensor const&, at::Type const&, at::optional<int>, bool) /var/lib/jenkins/workspace/torch/csrc/utils/tensor_conversion_dispatch.cpp:47:5
19:27:44     #11 0x7f24df633aca in torch::autograd::THPVariable_type(_object*, _object*, _object*) /var/lib/jenkins/workspace/torch/csrc/autograd/generated/python_variable_methods.cpp:632:27
19:27:44     #12 0x556394573b93 in _PyCFunction_FastCallDict (/opt/conda/bin/python3.6+0x10eb93)
19:27:44     #13 0x55639460367b in call_function (/opt/conda/bin/python3.6+0x19e67b)
19:27:44     #14 0x556394625cb9 in _PyEval_EvalFrameDefault (/opt/conda/bin/python3.6+0x1c0cb9)
19:27:44     #15 0x5563945fd70a in fast_function (/opt/conda/bin/python3.6+0x19870a)
19:27:44     #16 0x556394603754 in call_function (/opt/conda/bin/python3.6+0x19e754)
19:27:44     #17 0x556394625cb9 in _PyEval_EvalFrameDefault (/opt/conda/bin/python3.6+0x1c0cb9)
19:27:44     #18 0x5563945fd70a in fast_function (/opt/conda/bin/python3.6+0x19870a)
19:27:44     #19 0x556394603754 in call_function (/opt/conda/bin/python3.6+0x19e754)
19:27:44     #20 0x556394625cb9 in _PyEval_EvalFrameDefault (/opt/conda/bin/python3.6+0x1c0cb9)
19:27:44     #21 0x5563945fca93 in _PyEval_EvalCodeWithName (/opt/conda/bin/python3.6+0x197a93)
19:27:44     #22 0x5563945fe03a in _PyFunction_FastCallDict (/opt/conda/bin/python3.6+0x19903a)
19:27:44     #23 0x556394573f5e in _PyObject_FastCallDict (/opt/conda/bin/python3.6+0x10ef5e)
19:27:44     #24 0x556394578a02 in _PyObject_Call_Prepend (/opt/conda/bin/python3.6+0x113a02)
19:27:44     #25 0x55639457399d in PyObject_Call (/opt/conda/bin/python3.6+0x10e99d)
19:27:44     #26 0x55639462746f in _PyEval_EvalFrameDefault (/opt/conda/bin/python3.6+0x1c246f)
19:27:44     #27 0x5563945fca93 in _PyEval_EvalCodeWithName (/opt/conda/bin/python3.6+0x197a93)
19:27:44     #28 0x5563945fde1a in _PyFunction_FastCallDict (/opt/conda/bin/python3.6+0x198e1a)
19:27:44     #29 0x556394573f5e in _PyObject_FastCallDict (/opt/conda/bin/python3.6+0x10ef5e)
19:27:44     #30 0x556394578a02 in _PyObject_Call_Prepend (/opt/conda/bin/python3.6+0x113a02)
19:27:44     #31 0x55639457399d in PyObject_Call (/opt/conda/bin/python3.6+0x10e99d)
19:27:44     #32 0x5563945d09b6 in slot_tp_call (/opt/conda/bin/python3.6+0x16b9b6)
19:27:44     #33 0x556394573d7a in _PyObject_FastCallDict (/opt/conda/bin/python3.6+0x10ed7a)
19:27:44     #34 0x5563946037cd in call_function (/opt/conda/bin/python3.6+0x19e7cd)
19:27:44     #35 0x556394625cb9 in _PyEval_EvalFrameDefault (/opt/conda/bin/python3.6+0x1c0cb9)
19:27:44     #36 0x5563945fca93 in _PyEval_EvalCodeWithName (/opt/conda/bin/python3.6+0x197a93)
19:27:44     #37 0x5563945fe03a in _PyFunction_FastCallDict (/opt/conda/bin/python3.6+0x19903a)
19:27:44     #38 0x556394573f5e in _PyObject_FastCallDict (/opt/conda/bin/python3.6+0x10ef5e)
19:27:44     #39 0x556394578a02 in _PyObject_Call_Prepend (/opt/conda/bin/python3.6+0x113a02)
19:27:44     #40 0x55639457399d in PyObject_Call (/opt/conda/bin/python3.6+0x10e99d)
19:27:44     #41 0x55639462746f in _PyEval_EvalFrameDefault (/opt/conda/bin/python3.6+0x1c246f)
19:27:44     #42 0x5563945fca93 in _PyEval_EvalCodeWithName (/opt/conda/bin/python3.6+0x197a93)
19:27:44     #43 0x5563945fde1a in _PyFunction_FastCallDict (/opt/conda/bin/python3.6+0x198e1a)
19:27:44     #44 0x556394573f5e in _PyObject_FastCallDict (/opt/conda/bin/python3.6+0x10ef5e)
19:27:44     #45 0x556394578a02 in _PyObject_Call_Prepend (/opt/conda/bin/python3.6+0x113a02)
19:27:44     #46 0x55639457399d in PyObject_Call (/opt/conda/bin/python3.6+0x10e99d)
19:27:44     #47 0x5563945d09b6 in slot_tp_call (/opt/conda/bin/python3.6+0x16b9b6)
19:27:44     #48 0x556394573d7a in _PyObject_FastCallDict (/opt/conda/bin/python3.6+0x10ed7a)
19:27:44     #49 0x5563946037cd in call_function (/opt/conda/bin/python3.6+0x19e7cd)
19:27:44     #50 0x556394625cb9 in _PyEval_EvalFrameDefault (/opt/conda/bin/python3.6+0x1c0cb9)
19:27:44     #51 0x5563945fca93 in _PyEval_EvalCodeWithName (/opt/conda/bin/python3.6+0x197a93)
19:27:44     #52 0x5563945fe03a in _PyFunction_FastCallDict (/opt/conda/bin/python3.6+0x19903a)
19:27:44     #53 0x556394573f5e in _PyObject_FastCallDict (/opt/conda/bin/python3.6+0x10ef5e)
19:27:44     #54 0x556394578a02 in _PyObject_Call_Prepend (/opt/conda/bin/python3.6+0x113a02)
19:27:44     #55 0x55639457399d in PyObject_Call (/opt/conda/bin/python3.6+0x10e99d)
19:27:44     #56 0x55639462746f in _PyEval_EvalFrameDefault (/opt/conda/bin/python3.6+0x1c246f)
19:27:44     #57 0x5563945fca93 in _PyEval_EvalCodeWithName (/opt/conda/bin/python3.6+0x197a93)
19:27:44     #58 0x5563945fde1a in _PyFunction_FastCallDict (/opt/conda/bin/python3.6+0x198e1a)
19:27:44     #59 0x556394573f5e in _PyObject_FastCallDict (/opt/conda/bin/python3.6+0x10ef5e)
19:27:44     #60 0x556394578a02 in _PyObject_Call_Prepend (/opt/conda/bin/python3.6+0x113a02)
19:27:44     #61 0x55639457399d in PyObject_Call (/opt/conda/bin/python3.6+0x10e99d)
19:27:44     #62 0x5563945d09b6 in slot_tp_call (/opt/conda/bin/python3.6+0x16b9b6)
19:27:44     #63 0x556394573d7a in _PyObject_FastCallDict (/opt/conda/bin/python3.6+0x10ed7a)
19:27:44     #64 0x5563946037cd in call_function (/opt/conda/bin/python3.6+0x19e7cd)
19:27:44     #65 0x556394625cb9 in _PyEval_EvalFrameDefault (/opt/conda/bin/python3.6+0x1c0cb9)
19:27:44     #66 0x5563945fd70a in fast_function (/opt/conda/bin/python3.6+0x19870a)
19:27:44     #67 0x556394603754 in call_function (/opt/conda/bin/python3.6+0x19e754)
19:27:44     #68 0x556394625cb9 in _PyEval_EvalFrameDefault (/opt/conda/bin/python3.6+0x1c0cb9)
19:27:44     #69 0x5563945fd70a in fast_function (/opt/conda/bin/python3.6+0x19870a)
19:27:44     #70 0x556394603754 in call_function (/opt/conda/bin/python3.6+0x19e754)
19:27:44     #71 0x556394625cb9 in _PyEval_EvalFrameDefault (/opt/conda/bin/python3.6+0x1c0cb9)
19:27:44     #72 0x5563945fcf20 in _PyEval_EvalCodeWithName (/opt/conda/bin/python3.6+0x197f20)
19:27:44     #73 0x5563945fe03a in _PyFunction_FastCallDict (/opt/conda/bin/python3.6+0x19903a)
19:27:44     #74 0x556394573f5e in _PyObject_FastCallDict (/opt/conda/bin/python3.6+0x10ef5e)
19:27:44     #75 0x556394578a02 in _PyObject_Call_Prepend (/opt/conda/bin/python3.6+0x113a02)
19:27:44     #76 0x55639457399d in PyObject_Call (/opt/conda/bin/python3.6+0x10e99d)
19:27:44     #77 0x5563945d002a in slot_tp_init (/opt/conda/bin/python3.6+0x16b02a)
19:27:44     #78 0x5563946039b6 in type_call (/opt/conda/bin/python3.6+0x19e9b6)
19:27:44     #79 0x556394573d7a in _PyObject_FastCallDict (/opt/conda/bin/python3.6+0x10ed7a)
19:27:44     #80 0x5563945fdc29 in _PyObject_FastCallKeywords (/opt/conda/bin/python3.6+0x198c29)
19:27:44     #81 0x5563946037cd in call_function (/opt/conda/bin/python3.6+0x19e7cd)
19:27:44     #82 0x556394626a79 in _PyEval_EvalFrameDefault (/opt/conda/bin/python3.6+0x1c1a79)
19:27:44     #83 0x5563945a4a2a in _PyFunction_FastCall (/opt/conda/bin/python3.6+0x13fa2a)
19:27:44     #84 0x556394603754 in call_function (/opt/conda/bin/python3.6+0x19e754)
19:27:44     #85 0x556394625cb9 in _PyEval_EvalFrameDefault (/opt/conda/bin/python3.6+0x1c0cb9)
19:27:44     #86 0x5563945fe458 in PyEval_EvalCodeEx (/opt/conda/bin/python3.6+0x199458)
19:27:44     #87 0x5563945ff1eb in PyEval_EvalCode (/opt/conda/bin/python3.6+0x19a1eb)
19:27:44     #88 0x5563946799a3 in run_mod (/opt/conda/bin/python3.6+0x2149a3)
19:27:44     #89 0x556394679da0 in PyRun_FileExFlags (/opt/conda/bin/python3.6+0x214da0)
19:27:44     #90 0x556394679fa3 in PyRun_SimpleFileExFlags (/opt/conda/bin/python3.6+0x214fa3)
19:27:44     #91 0x55639467da9d in Py_Main (/opt/conda/bin/python3.6+0x218a9d)
19:27:44     #92 0x5563945454bd in main (/opt/conda/bin/python3.6+0xe04bd)
19:27:44     #93 0x7f24f839582f in __libc_start_main /build/glibc-Cl5G7W/glibc-2.23/csu/../csu/libc-start.c:291
19:27:44     #94 0x55639462c772 in _start (/opt/conda/bin/python3.6+0x1c7772)
19:27:44 
19:27:44 SUMMARY: UndefinedBehaviorSanitizer: undefined-behavior 
/var/lib/jenkins/workspace/aten/src/TH/generic/THTensorCopy.cpp:234:1 in 
19:27:44 Traceback (most recent call last):
19:27:44   File "test/run_test.py", line 345, in <module>
19:27:44     main()
19:27:44   File "test/run_test.py", line 337, in main
19:27:44     raise RuntimeError(message)
19:27:44 RuntimeError: test_torch failed!
```

It seems #9431 was not enough. CC @yf225

### PR DESCRIPTION
**Pull request by yf225**
Fixes https://github.com/pytorch/pytorch/issues/9554.

### CODE REVIEW

**Review by ssnl — APPROVED** [review]

**Comment by ezyang** [comment]
```
23:09:10 ======================================================================
23:09:10 ERROR: test_neg (test_torch.TestTorch)
23:09:10 ----------------------------------------------------------------------
23:09:10 Traceback (most recent call last):
23:09:10   File "/var/lib/jenkins/workspace/test/test_torch.py", line 1153, in test_neg
23:09:10     self._test_neg(self, lambda t: t)
23:09:10   File "/var/lib/jenkins/workspace/test/test_torch.py", line 1134, in _test_neg
23:09:10     zeros = cast(torch.Tensor(dtype=t.dtype)).resize_as_(a).zero_()
23:09:10 TypeError: new() received an invalid combination of arguments - got (dtype=torch.dtype, ), but expected one of:
23:09:10  * (torch.device device)
23:09:10       didn't match because some of the keywords were incorrect: dtype
23:09:10  * (torch.Storage storage)
23:09:10       didn't match because some of the keywords were incorrect: dtype
23:09:10  * (Tensor other)
23:09:10       didn't match because some of the keywords were incorrect: dtype
23:09:10  * (tuple of ints size, torch.device device)
23:09:10  * (object data, torch.device device)
```

File: test/test_torch.py [hunk 1]
```diff
@@ -1068,18 +1068,18 @@ def test_csub(self):
...
+        float_types = [torch.DoubleTensor, torch.FloatTensor, torch.LongTensor]
+        int_types = [torch.IntTensor, torch.ShortTensor, torch.ByteTensor,
+                     torch.CharTensor]
 
         for t in float_types + int_types:
             if t in float_types:
                 a = cast(torch.randn(100, 90).type(t))
             else:
-                a = cast(torch.Tensor(100, 90).type(t).random_(-128, 128))
+                a = cast(torch.Tensor(100, 90, dtype=t.dtype).random_(-128, 128))
```
Comment (colesbury) [inline test/test_torch.py]: `a = cast(torch.randint(-128, 128, (100, 900), dtype=t.dtype))`

**Pull request closed without merging**